include/xrpl/resource/detail/Logic.h

```diff
@@ -442,12 +442,33 @@ class Logic
     }
 
     Disposition
-    charge(Entry& entry, Charge const& fee)
+    charge(Entry& entry, Charge const& fee, std::string context = {})
     {
+        static constexpr Charge::value_type feeLogAsWarn = 3000;
+        static constexpr Charge::value_type feeLogAsInfo = 1000;
+        static constexpr Charge::value_type feeLogAsDebug = 100;
+        static_assert(
+            feeLogAsWarn > feeLogAsInfo && feeLogAsInfo > feeLogAsDebug &&
+            feeLogAsDebug > 10);
+
+        auto getStream = [this](Resource::Charge::value_type cost) {
+            if (cost >= feeLogAsWarn)
+                return m_journal.warn();
+            if (cost >= feeLogAsInfo)
+                return m_journal.info();
+            if (cost >= feeLogAsDebug)
+                return m_journal.debug();
+            return m_journal.trace();
+        };
```

Comment:
Not necessary for this PR, but `getStream` could probably be changed to static if you remove don't capture `this`, and pass in a `Journal&`. Then the call would be `getStream(m_journal, fee.cost())`. Then again, the compiler probably optimizes the whole thing away anyway.